### PR TITLE
Update contributing docs

### PR DESCRIPTION
**Pull request by waynew**
### What does this PR do?

Updates the contributing docs to be in-line with the `master` branch strategy, as well as reference `pre-commit`

### What issues does this PR fix or reference?

A couple of SEPs, comments in Slack, and other places.

### Tests written?

Doc changes only

### Commits signed with GPG?

Yes

---

Note: this PR is waiting for https://github.com/saltstack/salt/pull/55765 to be merged, so contributors can really use black.

### CODE REVIEW

File: doc/topics/development/contributing.rst [hunk 1]
```diff
@@ -209,8 +247,8 @@ Fork a Repo Guide_>`_ and is well worth reading.
 
     Pull request against `saltstack/salt`_ are automatically tested on a
     variety of operating systems and configurations. On average these tests
-    take 30 minutes.  Depending on your GitHub notification settings you may
-    also receive an email message about the test results.
+    take a couple of hours.  Depending on your GitHub notification settings
+    you may also receive an email message about the test results.
```
Comment (s0undt3ch) [inline doc/topics/development/contributing.rst]: We send out emails through GH? 
I don't think we do, do we?

**Review by s0undt3ch — APPROVED** [review]

File: doc/topics/development/contributing.rst (hunk 1 above)
Comment (waynew) [reply doc/topics/development/contributing.rst]: GH might send emails - we don't.

Comment (waynew) [reply doc/topics/development/contributing.rst]: (they [can] send a bunch of different notifications depending on what your settings are)

Comment (waynew) [reply doc/topics/development/contributing.rst]: (they [can] send a bunch of different notifications depending on what your settings are)

**Comment by waynew** [comment]
TODO: add a note about copyright/license headers (not needed/allowed)

**Review by Akm0d — APPROVED** [review]

**Pull request merged**